### PR TITLE
[Welcome] Reorganization of command groups

### LINKED ISSUE #533: [Welcome] Reorganization of command groups
## Summary

As mentioned in #520, we should restructure the cog's command groups to make them look more organized.

## Details

Currently, we have

`[p]welcomeset`
* `dm`/`toggledm`
* `greetings`
    * `add`
    * `channel`
    * `list`
    * `remove`
* `image`
* `log`/`togglelog`
* `logchannel`
* `message`/`msg`
* `tag`/`desc`/`description`/`descriptions`
    * `add`
    * `get`
    * `list`
    * `remove`
* `test`
* `title`

We want them to be

`[p]welcomeset`
* `dm`
    * `message`
    * `title`
    * `image`
    * `test`
    * `toggle`
* `greetings`
    * `add`
    * `channel`
    * `list`
    * `remove`
    * `toggle`
* `log`
    * `channel`
    * `toggle`
* `tag`/`desc`/`description`/`descriptions`
    * `add`
    * `get`
    * `list`
    * `remove`

## Notes

We usually store config values in `guildConfig[KEY_X][KEY_Y]` when we have command group `Y` being under the command group `X`, however, I think it is a good idea to just change the command group structure and leave the config data structure as is for now. If needed, we can come back to it in another issue.

### PR DESCRIPTION
**Pull request by quachtridat**
## Summary

- Resolve #533, except there is no `[p]welcomeset greetings toggle`. We can have another PR for that.

- No changes in implementation of any methods. Changes are only made for `Red`-related method decorators, method declarations, docstring, and the order these methods appear in the `cogs/welcome/welcome.py` file.

### CODE REVIEW

File: cogs/welcome/welcome.py [hunk 1]
```diff
@@ -252,17 +252,145 @@ async def logServerLeave(self, leaveUser: discord.Member):
...
+            return
+
+        await self.config.guild(ctx.guild).get_attr(KEY_TITLE).set(title.content)
+        await ctx.send("Title set to:")
+        await ctx.send(f"```{title.content}```")
+        LOGGER.info(
+            "Title changed by %s#%s (%s)",
+            ctx.message.author.name,
+            ctx.message.author.discriminator,
+            ctx.message.author,
```
Comment (Injabie3) [inline cogs/welcome/welcome.py]: LOL, can you fix this to `ctx.message.author.id`?

File: cogs/welcome/welcome.py [hunk 2]
```diff
@@ -252,17 +252,145 @@ async def logServerLeave(self, leaveUser: discord.Member):
...
+    async def setImage(self, ctx: Context, imageUrl: str = None):
+        """Sets an image in the embed with a URL.
+
+        Parameters:
+        -----------
+        imageUrl: str (optional)
+            The URL of the image to use in the DM embed. Leave blank to disable.
+        """
+        if imageUrl == "":
+            imageUrl = None
```
Comment (Injabie3) [inline cogs/welcome/welcome.py]: Naive question: is this when someone explicitly does `[p]welcomeset dm image ""`? Can you double check and see if we can remove this?

File: cogs/welcome/welcome.py [hunk 3]
```diff
@@ -252,17 +252,145 @@ async def logServerLeave(self, leaveUser: discord.Member):
...
+        await self.config.guild(ctx.guild).get_attr(KEY_IMAGE).set(imageUrl)
+        if imageUrl:
+            await ctx.send(f"Welcome image set to `{imageUrl}`. Be sure to test it!")
+        else:
+            await ctx.send("Welcome image disabled.")
+        LOGGER.info(
+            "Image changed by %s#%s (%s)",
+            ctx.message.author.name,
+            ctx.message.author.discriminator,
+            ctx.message.id,
```
Comment (Injabie3) [inline cogs/welcome/welcome.py]: Can you also fix `ctx.message.author.id`?

File: cogs/welcome/welcome.py [hunk 4]
```diff
@@ -364,58 +470,29 @@ def check(message: discord.Message):
...
-        name: str
-            Name of the greeting to remove
-        pool: str
-            A greeting pool to remove from; leave blank for the default pool
-            - `default`: default pool, containing greetings that are sent to new users
-            - `returning`: pool of greetings that are sent to returning users
+        channel: discord.TextChannel
+            The text channel to set welcome's to. If not passed anything, will remove the welcome channel
         """
+        if not channel:
```
Comment (Injabie3) [inline cogs/welcome/welcome.py]: I know this was moved, but can we change this to `if channel is None:`?

**Review by Injabie3 — CHANGES_REQUESTED** [review]
Thanks, I know this was pretty much just moving stuff around, but can you fix the following in the next commit?

File: cogs/welcome/welcome.py (hunk 2 above)
Comment (quachtridat) [reply cogs/welcome/welcome.py]: We can remove this block.

**Review by Injabie3 — APPROVED** [review]
Thanks, LGTM

**Pull request merged**